macro/Cargo.toml

```diff
@@ -16,6 +16,7 @@ description = "Generate types and helpers for interacting with Substrate runtime
 [features]
 web = ["subxt-codegen/web"]
 runtime-path = ["polkadot-sdk"]
+jsonrpsee = ["subxt-codegen/jsonrpsee"]
```

Comment:
I'd change this to `fetch-metadata = ["subxt-codegen/fetch-metadata"]` and ditch the `jsonrpsee` flag I reckon.

Then, whether you can fetch metadata from a URL or not depends on this feature being enabled, rather than us always bringing in jsonrpsee (at compile time) to allow that in the macro.

I think this is all that actually needs doing in this PR to remove jsonrpsee; no need for the etra `jsonrpsee` feature flag etc I think?

Reply:
(should also be noted that jsonrpsee only comes in at compile time for the macro to run, so I don't think it's awfully important here. The thing I would want to avoid is jsonrpsee being bought in as a runtime dependency evcen when we don't enable the `jsonrpsee` feature in subxt)

Reply:
I think I rather breakout the fetch-metadata to a separate crate to simplify this

Reply:
Yeah we could do; it just always felt like overkill having a separate crate for this one thing!

I think at the mo I prefer just making `fetch-metadata` an optional feature on the macro crate and then that's all we really need to do here I think?!

Reply:
That said, we could consider adding a `utils` folder which contains some of the little crates we might end up wanting, and then one of those crates is `fetch-metadata` which contains the code to fetch metadata from URLs and is only imported in the macro if the `fetch-metadata` feature is enabled?